src/promptflow-evals/promptflow/evals/evaluators/_xpia/_xpia.py

```diff
@@ -0,0 +1,305 @@
+                "content": "2 + 2 = 4",
+                "context": {
+                    "citations": [{"id": "doc.md", "content": "Information about additions: 1 + 2 = 3, 2 + 2 = 4"}]
+                },
+            },
+            {"role": "user", "content": "What is the capital of Japan?"},
+            {
+                "role": "assistant",
+                "content": "The capital of Japan is Tokyo.",
+                "context": {
+                    "citations": [
+                        {
+                            "id": "doc.md",
+                            "content": "Tokyo is Japan's capital, known for its blend of traditional culture and \
+                                technological"
+                            "advancements.",
+                        }
+                    ]
+                },
+            },
+        ]
+        result = eval_fn(conversation=conversation)
+
+    **Output format for conversation**
+
+    .. code-block:: python
+
+        {
+            'label': False,
+            'evaluation_per_turn':
+                {
+                    'label': [False, False],
+                    'reasoning': ['The conversation does not contain any manipulated content,
+                    intrusion or information gathering..', 'The conversation does not contain
+                    any manipulated content, intrusion or information gathering..']
+                }
+        }
+
+    **Output format for question-answer pair**
+
+    .. code-block:: python
+
+        {
+            'label': False,
+            'reasoning': 'The conversation does not contain any manipulated content, intrusion or
+                          information gathering..'
+        }
+
+    """
+
+    def __init__(self, project_scope: dict, eval_last_turn: bool = False, credential=None):
+        self._evaluator = _IndirectAttackEvaluator(project_scope, credential)
+        self._eval_last_turn = eval_last_turn
+
+    def __call__(
+        self,
+        *,
+        conversation=None,
+        question: Optional[str] = None,
+        answer: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        Evaluates content according to the presence of attacks injected into the conversation context to
+        interrupt normal expected functionality by eliciting manipulated content, intrusion and attempting
+        to gather information outside the scope of your AI system.
+
+        :keyword conversation: The conversation to be evaluated. Each turn should have "role" and "content" keys.
+        :paramtype conversation: Optional[List[Dict]]
+        :keyword question: The question to be evaluated. Mutually exclusive with 'conversation'.
+        :paramtype question: Optional[str]
+        :keyword answer: The answer to be evaluated. Mutually exclusive with 'conversation'.
+        :paramtype answer: Optional[str]
+        :return: The evaluation scores and reasoning.
+        :rtype: dict
+        """
+        if conversation:
```

Comment:
If user provides both conversation and (question, answer) does it need some validation ?

Reply:
Raising an error would be better since that makes it explicit for users.

Reply:
done